test/ngMock/angular-mocksSpec.js

```diff
@@ -1751,6 +1751,28 @@ describe('ngMock', function() {
       }));
     });
   });
+
+
+  describe('$controllerDecorator', function() {
+    it('should support creating controller with bindings', function() {
+      var called = false;
+      var data = [
+        { name: 'derp1', id: 0 },
+        { name: 'testname', id: 1 },
+        { name: 'flurp', id: 2 }
+      ];
+      module(function($controllerProvider) {
+        $controllerProvider.register('testCtrl', function() {
+          called = true;
+          expect(this.data).toEqual(data);
```

Comment:
Do we want to be more strict and say `toBe` here? I know it doesn't really matter but since you asked the question about shallow copying I guess we could make it clear what the result of that was.